src/main/java/control/DirectorTimelineController.java

```diff
@@ -1,16 +1,187 @@
 package control;
 
+import data.DirectorShot;
+import data.DirectorTimeline;
+import data.Shot;
+import gui.centerarea.CameraShotBlock;
+import gui.centerarea.DirectorShotBlock;
+import gui.events.DirectorShotBlockUpdatedEvent;
+import gui.root.RootPane;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
 import lombok.Getter;
+import lombok.extern.log4j.Log4j2;
+
 
 /**
  * Controller for the DirectorTimeline.
  */
+@Log4j2
 public class DirectorTimelineController {
 
+    private RootPane rootPane;
+
     @Getter
     private final ControllerManager controllerManager;
 
+    @Getter
+    // List of all DirectorShotBlocks in this timeline
+    private ArrayList<DirectorShotBlock> shotBlocks;
+
+    // List of all currently colliding DirectorShotBlocks
+    private ArrayList<DirectorShotBlock> overlappingShotBlocks;
+
+    /**
+     * Constructor.
+     * @param controllerManager Root controller
+     */
     public DirectorTimelineController(ControllerManager controllerManager) {
+        log.debug("Constructing new TimelineController(controllerManager={})", controllerManager);
+
         this.controllerManager = controllerManager;
+        this.rootPane = controllerManager.getRootPane();
+        this.shotBlocks = new ArrayList<>();
+        this.overlappingShotBlocks = new ArrayList<>();
+    }
+
+    /**
+     * Add a director shot to the corresponding timeline.
+     * @param name Name of the shot
+     * @param description Shot description
+     * @param startCount Start count
+     * @param endCount End count
+     * @param frontPadding additional recording time at the front
+     * @param endPadding additional recording time at the end
+     * @param cameras the cameras to generate the shot to
+     */
+    public void addDirectorShot(String name, String description,
+                                double startCount, double endCount,
+                                double frontPadding, double endPadding,
+                                List<Integer> cameras) {
+        log.info("Adding DirectorShot to DirectorTimeline");
+
+        DirectorShot newShot = new DirectorShot(name, description, startCount, endCount,
+                frontPadding, endPadding, cameras);
+        this.controllerManager.getScriptingProject()
+                .getDirectorTimeline()
+                .addShot(newShot);
+        DirectorShotBlock shotBlock = new DirectorShotBlock(newShot.getInstance(),
+                rootPane.getRootCenterArea(), startCount, endCount,
+                description, name, this::shotChangedHandler, newShot);
+
+        controllerManager.setActiveShotBlock(shotBlock);
+        this.shotBlocks.add(shotBlock);
+        controllerManager.getScriptingProject().changed();
+
+        // Check for collisions
+        checkCollisions(shotBlock);
+    }
+
+    /**
+     * Handle updated director shot. The previous timeline is used to retrieve the corresponding
```

Comment:
There is no switching of timelines going on when drag-and-droping directorshots. So please remove the parts of the javadoc that states that